src/lib/Serializer/SuggestionSerializer.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Search\Serializer;
+
+use Symfony\Component\Serializer\Encoder\JsonEncoder;
+use Symfony\Component\Serializer\Normalizer\AbstractNormalizer;
+use Symfony\Component\Serializer\Normalizer\ObjectNormalizer;
+use Symfony\Component\Serializer\Serializer;
+use Symfony\Component\Serializer\SerializerInterface;
+
+final class SuggestionSerializer implements SerializerInterface
+{
+    private SerializerInterface $serializer;
+
+    public function __construct()
+    {
+        $encoders = [new JsonEncoder()];
+
+        $defaultContext = [
+            AbstractNormalizer::CIRCULAR_REFERENCE_HANDLER => static function ($object) {
+                return $object->getId();
+            },
+        ];
+
+        $normalizers = [new ObjectNormalizer(null, null, null, null, null, null, $defaultContext)];
```

Comment:
I tend to stay away from `ObjectNormalizer` in favor of creating a dedicated normalizer for each class I am working with. This is to ensure the specific shape of the response, not reliant on `ObjectNormalizer` picking properties out of an object.